### PR TITLE
[blog] 2023-10-25 implementing sign in with google

### PR DESCRIPTION
**Pull request by masenf**
@Alek99 can you midjourney me a sick cover image for this post?

Read it first: https://pcweb-red-ring.reflex.run/blog/2023-10-25-implementing-sign-in-with-google/

### CODE REVIEW

File: blog/2023-10-25-implementing-sign-in-with-google.md [hunk 1]
```diff
@@ -0,0 +1,519 @@
...
+
+Define the following wrapper in the same `react_oauth_google.py`.
+
+```python
+class GoogleLogin(rx.Component):
+    library = "@react-oauth/google"
+    tag = "GoogleLogin"
+
+    def get_event_triggers(self):
+        return \{"on_success": lambda data: [data]}
```
Comment (tgberkeley) [inline blog/2023-10-25-implementing-sign-in-with-google.md:102]: Not 100% clear how the `on_success` trigger here relates to the `on_success` State function defined on line 145

File: blog/2023-10-25-implementing-sign-in-with-google.md [hunk 2]
```diff
@@ -0,0 +1,519 @@
...
+            client_id=CLIENT_ID
+        )
+    )
+
+app = rx.App()
+app.add_page(index)
+app.compile()
+```
+
+After a successful login, you will see the decoded JWT with user profile
```
Comment (tgberkeley) [inline blog/2023-10-25-implementing-sign-in-with-google.md:180]: Might not be obvious what JWT is for a beginner

File: blog/2023-10-25-implementing-sign-in-with-google.md (hunk 1 above)
Comment (masenf) [reply blog/2023-10-25-implementing-sign-in-with-google.md:102]: There is a line in the frontend code that associates them

```python
GoogleLogin.create(on_success=State.on_success)
```

File: blog/2023-10-25-implementing-sign-in-with-google.md [hunk 3]
```diff
@@ -0,0 +1,519 @@
...
+        if self.token_is_valid:
+            return f"This content can only be viewed by a logged in User. Nice to see you \{self.tokeninfo['name']}"
+        return "Not logged in."
+```
+
+```python eval
+docalert(
+    "Private data must come from the State",
+    "Reflex compiles the page function and makes its content publicly "
+    "accessible! Although conditional rendering is used to to show the "
```
Comment (tgberkeley) [inline blog/2023-10-25-implementing-sign-in-with-google.md:332]: two "to"s here on this line

File: blog/2023-10-25-implementing-sign-in-with-google.md [hunk 4]
```diff
@@ -0,0 +1,519 @@
...
+information displayed in the terminal!
+
+## Saving the Token
+
+The `GoogleLogin` component does NOT store the token in any way, so it is up to
+our app to store and manage the credential after login. For this purpose, we
+will use an `rx.LocalStorage` `Var` in the `State` that is set in the
+`on_success` event handler.
+
+Additionally, an `rx.cached_var` will be used to verify and return the decoded
```
Comment (tgberkeley) [inline blog/2023-10-25-implementing-sign-in-with-google.md:190]: could be a link to cached_var in the docs

File: blog/2023-10-25-implementing-sign-in-with-google.md [hunk 5]
```diff
@@ -0,0 +1,519 @@
...
+```
+
+After a successful login, you will see the decoded JWT with user profile
+information displayed in the terminal!
+
+## Saving the Token
+
+The `GoogleLogin` component does NOT store the token in any way, so it is up to
+our app to store and manage the credential after login. For this purpose, we
+will use an `rx.LocalStorage` `Var` in the `State` that is set in the
```
Comment (tgberkeley) [inline blog/2023-10-25-implementing-sign-in-with-google.md:187]: could be a link to LocalStorage in the docs

File: blog/2023-10-25-implementing-sign-in-with-google.md (hunk 4 above)
Comment (masenf) [reply blog/2023-10-25-implementing-sign-in-with-google.md:190]: if there was a section on `cached_var` in the docs 😬

Comment (tgberkeley) [reply blog/2023-10-25-implementing-sign-in-with-google.md:190]: ah if not then this itself needs to be a ticket

File: blog/2023-10-25-implementing-sign-in-with-google.md (hunk 1 above)
Comment (tgberkeley) [reply blog/2023-10-25-implementing-sign-in-with-google.md:102]: Ah yes I added this comment before I got down to this line in the code

File: blog/2023-10-25-implementing-sign-in-with-google.md (hunk 4 above)
Comment (masenf) [reply blog/2023-10-25-implementing-sign-in-with-google.md:190]: REF-1019

Comment (tgberkeley) [reply blog/2023-10-25-implementing-sign-in-with-google.md:190]: created a linear ticket for this

File: blog/2023-10-25-implementing-sign-in-with-google.md (hunk 2 above)
Comment (masenf) [reply blog/2023-10-25-implementing-sign-in-with-google.md]: added a link to jwt.io and expanded the acronym inline

File: blog/2023-10-25-implementing-sign-in-with-google.md (hunk 1 above)
Comment (masenf) [reply blog/2023-10-25-implementing-sign-in-with-google.md:104]: do you think it's okay, or should i name the state event handler something else, like `handle_token`?

**Comment by Alek99** [comment]
> @Alek99 can you midjourney me a sick cover image for this post?
> 
> Read it first: https://pcweb-red-ring.reflex.run/blog/2023-10-25-implementing-sign-in-with-google/

Ill do this now

**Review by Alek99 — APPROVED** [review]
Look good nice! 👍

**Pull request merged**